### PR TITLE
[FIX] Fix speech2test modeling tests

### PR DESCRIPTION
**Pull request by ylacombe**
# What does this PR do?

Fixes the modeling tests of speech2text.
I've also run the slow tests of course!

cc @ydshieh and @amyeroberts

### CODE REVIEW

File: tests/models/speech_to_text/test_modeling_speech_to_text.py [hunk 1]
```diff
@@ -284,6 +284,16 @@ class Speech2TextModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTest
 
     input_name = "input_features"
 
+    def _get_input_ids_and_config(self, batch_size=2):
+        config, input_ids, attention_mask, max_length = GenerationTesterMixin._get_input_ids_and_config(self)
+
+        # overwrite because otherwise the attention mask'd have the same shape than the inputs here, which is a 3D tensor.
```
Comment (ydshieh) [inline tests/models/speech_to_text/test_modeling_speech_to_text.py:290]: if not too much work, might be nice to explain why it is 3D (it seems this is special situation to this model?) and why we need 2D.

**Review by ydshieh — APPROVED** [review]
Thanks for the quick fix

File: tests/models/speech_to_text/test_modeling_speech_to_text.py [hunk 2]
```diff
@@ -284,6 +284,18 @@ class Speech2TextModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTest
 
     input_name = "input_features"
 
+    def _get_input_ids_and_config(self, batch_size=2):
+        config, input_ids, attention_mask, max_length = GenerationTesterMixin._get_input_ids_and_config(self)
+
+        # `input_ids` is actually `input_features` which is a 3D tensor.
+        # We must overwrite the mask to make it 2D since the original `_get_input_ids_and_config` creates an
+        # attention mask of the same shape than `input_ids`.
```
Comment (amyeroberts) [inline tests/models/speech_to_text/test_modeling_speech_to_text.py:292]: ultra ultra nit 

```suggestion
        # attention mask of the same shape as `input_ids`.
```

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

**Pull request merged**